### DISCUSSION TITLE
DV - R - Cara memperbaiki anotasi label overlap pada ggplot dari fungsi geom_text

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#58] [Q&A] [answered]
Dengan code berikut saya mencoba menandai timeline tanggal dengan rilisan lagu, namun bagaimana cara agar anotasi tersebut tidak tumpang tindih? 
```
ggplot(timelines_marked, aes(x=date, y=followers))+
  geom_line()+
  geom_vline(data=recordings_marker, 
             aes(xintercept=release_date),
             linetype="dashed")+
  geom_text(data=recordings_marker,
            aes(x=release_date,
                label=song),
            y=8,
           )
```
![overlapping labels](https://user-images.githubusercontent.com/95233522/164434095-a2ff81d3-7d7d-488b-a1ab-51f0d887616a.JPG)
Sebelumnya menemukan jawaban dari stackoverflow dibilang agar menambahkan position_jitter, sudah saya coba namun malah error.
```
Error in `check_required_aesthetics()`:
! position_jitter requires the following missing aesthetics: y
Backtrace:
  1. base `<fn>`(x)
  2. ggplot2:::print.ggplot(x)
  4. ggplot2:::ggplot_build.ggplot(x)
  5. ggplot2 by_layer(function(l, d) l$compute_position(d, layout))
  6. ggplot2 f(l = layers[[i]], d = data[[i]])
  7. l$compute_position(d, layout)
  8. ggplot2 f(..., self = self)
  9. self$position$setup_data(data, params)
 10. ggplot2 f(..., self = self)
 11. ggplot2:::check_required_aesthetics(...)
 Error in check_required_aesthetics(self$required_aes, names(data), snake_class(self)) 
```

**Comment by tomytjandra** [comment, marked as answer]
Halo Pak @Hertzspr, berikut adalah alternatif jawaban yang bisa saya berikan untuk mengatur `geom_text()` agar tidak tumpang tindih:

1. Menambahkan parameter `position` di dalam `geom_text()` serta memindahkan parameter `y` ke dalam `aes()` karena `position_jitter()` karena fungsi tersebut memutuhkan aesthetic `y`. Sebagai contoh:
```
geom_text(data=recordings_marker, aes(x=release_date, y=8, label=song))
```

2. Menggunakan fungsi `geom_text_repel()` dari library `ggrepel`. Sebagai contoh:
```
geom_text_repel(data=recordings_marker, aes(x=release_date, label=song), y=8)
```

**Catatan:** Apabila saat zoom in atau preview plot RStudio tidak muncul textnya, silahkan knit menjadi HTML untuk melihat hasil akhir dari plotnya.

Semoga membantu 😄

**Reply by Hertzspr** [reply]
Terima kasih mas Tomy
